modules/core/src/main/scala/shop/programs/checkout.scala

```diff
@@ -51,15 +51,17 @@ final class CheckoutProgram[F[_]: Background: Logger: MonadThrow: Timer](
       onError = logError("Order")
     )(orders.create(userId, paymentId, items, total))
 
-    action
-      .adaptError {
-        case e => OrderError(e.getMessage)
-      }
-      .onError {
-        case _ =>
-          Logger[F].error(s"Failed to create order for Payment: ${paymentId}. Rescheduling as a background action") *>
-              Background[F].schedule(action, 1.hour)
-      }
+    def bgAction(fa: F[OrderId]): F[OrderId] =
+      fa.adaptError {
+          case e => OrderError(e.getMessage)
+        }
+        .onError {
+          case _ =>
+            Logger[F].error(s"Failed to create order for Payment: ${paymentId}. Rescheduling as a background action") *>
+                Background[F].schedule(fa, 1.hour)
```

Comment:
Are you sure this actually changes anything? Wouldn't this line have to be `Background[F].schedule(bgAction(fa), 1.hour)`?

Reply:
Damn, you're right! :man_facepalming: